### PR TITLE
WIP: Dual stack support for name space address sets

### PR DESCRIPTION
**Pull request by gvbalaji**
Dual stack support for adding address sets to Namespaces
    
    1. Support for adding and deleting dual stack POD IP addresses to Namespace address sets.
    2. Support for dual stack IP addresses while adding Namespace.

### CODE REVIEW

File: go-controller/pkg/ovn/namespace.go [hunk 1]
```diff
@@ -7,14 +7,18 @@ import (
...
 	kapi "k8s.io/api/core/v1"
 	utilwait "k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/klog"
+	utilnet "k8s.io/utils/net"
 )
 
 const (
 	// Annotation used to enable/disable multicast in the namespace
 	nsMulticastAnnotation = "k8s.ovn.org/multicast-enabled"
+	iPv4AddressSetSuffix  = "_v4"
```
Comment (danwinship) [inline go-controller/pkg/ovn/namespace.go:20]: golang capitalization style is that if the first letter of an acronym has to be lowercase then the whole acronym becomes lowercase. So `ipv4AddressSetSuffix` with a lowercase "p".

Likewise, if the first letter is capitalized, then leave the whole acronym capitalized. so eg `v4Ips` below should be `v4IPs`

File: go-controller/pkg/ovn/namespace_test.go [hunk 2]
```diff
@@ -91,7 +95,7 @@ var _ = Describe("OVN Namespace Operations", func() {
 				_, err := fakeOvn.fakeClient.CoreV1().Namespaces().Get(namespaceT.Name, metav1.GetOptions{})
 				Expect(err).NotTo(HaveOccurred())
 
-				fakeOvn.asf.ExpectAddressSetWithIPs(namespaceT.Name, []string{tP.podIP})
+				fakeOvn.asf.ExpectAddressSetWithIPs(v4AddressSetName, []string{tP.podIP})
```
Comment (danwinship) [inline go-controller/pkg/ovn/namespace_test.go:98]: Should update these tests to expect a v6 address set to _not_ exist as well.

File: go-controller/pkg/ovn/policy.go [hunk 3]
```diff
@@ -892,15 +893,31 @@ func (oc *Controller) handlePeerNamespaceSelector(
 				np.Lock()
 				defer np.Unlock()
 				if !np.deleted {
-					gress.addNamespaceAddressSet(namespace.Name, np.portGroupName)
+					//FIXME :  Dual stack . We need to first fix ACL generation in case of dual stack
```
Comment (danwinship) [inline go-controller/pkg/ovn/policy.go:896]: (always put a space after `//`)

File: go-controller/pkg/ovn/policy.go [hunk 4]
```diff
@@ -892,15 +893,31 @@ func (oc *Controller) handlePeerNamespaceSelector(
...
+					// Then we will remove the if logic and add both address sets.
+					// Will be fixed in next set of PRs
+
+					if config.IPv6Mode {
+						gress.addNamespaceAddressSet(getIPv6AddressSetName(namespace.Name), np.portGroupName)
```
Comment (danwinship) [inline go-controller/pkg/ovn/policy.go:901]: So there's `config.IPv4Mode` too. What you want is to construct an ACL where in ipv4-only mode you check "`ip4.src == $IP4_ADDRESS_SET`" and in ipv6-only mode you check "`ip6.src == $IP6_ADDRESS_SET`" and in dual-stack mode (`config.IPv6Mode && config.IPv4Mode`) you check "`(ip4.src == $IP4_ADDRESS_SET || ip6.src == $IP6_ADDRESS_SET)`"

**Comment by gvbalaji** [comment]
thanks for the comments @danwinship . I'll incorporate them when I get back to this after making changes to AddressSet APIs as discussed.

**Pull request closed without merging**